tests/functionality_tests/test_activation_shapes.py

```diff
@@ -44,11 +71,22 @@ def main():
         y3 = cols // 6 ## Good value 5
         grid.activate_triangle(x1,y1,x2,y2,x3,y3) 
         grid.find_border_nodes()
-        way_points = grid.get_all_lawnmower_waypoints_adjustable()
+        way_points = grid.get_all_guided_lawnmower_waypoints_adjustable(is_vertical)
 
+    if activation_type == "line":
+        row = 0 # rows are y position
+        col = cols // 2 # cols are x position
+        grid.activate_line(row, col, n=15, isHorizontal=False)
```

Comment:
is n=15 something that works for all lines?

Reply:
I think this is hardcoded for a specific case of a vertical line going straight up by 15.

Reply:
^ any ideas?

Reply:
^^ maybe n = cols//2 for half the width of the grid? agree we should code it dynamically

Reply:
Changed, the lines are vertical going up, but our rows/columns are swapped, so it will be cols//2  for half the height of the grid